enhancements/proposals/1311-replace-nodes-using-hostid/README.md

```diff
@@ -0,0 +1,108 @@
+that user is using deprecated procedure and they should shift to the new one.
+
+### Risks and Mitigations
+
+#### ScyllaDB version requirement
+
+Node replacement procedure alignment requires the newest 5.2 ScyllaDB Open Source and 2023.1 Enterprise, meaning node
+replacement procedure won’t work when Scylla doesn’t meet minimum version criteria. Users wanting to update to a newer
+Operator version will need to update their ScyllaDB versions. We will mention it in release notes.
+
+## Design Details
+
+### API changes
+
+Status field dedicated to storing IP addresses of nodes being replaced is going to be deprecated.
+
+```go
+type RackStatus struct {
+    // DEPRECATED
+    // replace_address_first_boot holds addresses which should be replaced by new nodes.
+    ReplaceAddressFirstBoot map[string]string `json:"replace_address_first_boot,omitempty"`
+}
+```
+
+Field will be used until the old procedure is removed in at least the next version. After it’s gone, the map will always
+be empty.
+
+### Procedure flow
+
+#### Existing procedure
+
+Current replace procedure consists of putting a `scylla/replace` label on a node dedicated Service. Operator detects
+that
+the user wants to replace this node, stores the IP address of the node to be replaced in `ScyllaCluster.status.racks[]
+.replace_address_first_boot`.
+After that, Service, Pod and PVC are removed, and new Service is created with a label `scylla/replace` having the IP
+address of the node which is being replaced. Sidecar picks this value and propagates it to the
+`replace-address-first-boot` parameter to ScyllaDB. When Pod becomes Ready, the label is removed from the Service and
+status is cleared.
+
+#### New procedure
+
+Operator will trigger this procedure only when it detects it’s supported by the deployed ScyllaDB version, otherwise it
+will fallback to the old procedure.
+Instead of keeping HostID of a node being replaced in status, will keep the Service intact and sidecar will use HostID
+stored as an annotation to add a replace-node-first-boot parameter to ScyllaDB when the replace label is present.
+ScyllaDB supports replacing a node with the same IP address, this won’t disturb replacing when the Operator is being
+upgraded and versions are skewed.
+
+When a Scylla Operator sees that a replacement label `scylla/replace` is added by the user to the node dedicated Service
+and it’s value is empty, node Pod and associated PVC will be removed, and Service will be updated with
+`internal.scylla-operator.scylladb.com/replacing-node-hostid` with node Host ID as a value.
+When the new Pod comes up, sidecar will propagate the value of HostID annotation from Service into
+`replace-node-first-boot` parameter into ScyllaDB.
+When Service Host ID annotation finally mismatches value stored in the replacement label, and Pod becomes ready, the
+label will be removed ending the procedure.
+
+## Test Plan
+
+E2E tests will verify that both old and new procedures are replacing nodes.
```

Comment:
Please be a bit more explicit; is the current e2e test covering replacement going to stay as is?

Reply:
done